licenses/scripts/pull_licenses_py.py

```diff
@@ -0,0 +1,157 @@
+  dest_dir = '/'.join([license_dir, name.lower()])
+  try:
+    if not os.path.isdir(dest_dir):
+      os.mkdir(dest_dir)
+    shutil.copy(source_license_file, dest_dir + '/LICENSE')
+    return True
+  except Exception as e:
+    print(e)
+    return False
+
+
+@retry(stop=stop_after_attempt(3))
+def pull_from_url(dep, configs):
+  '''
+  :param dep: name of a dependency
+  :param configs: a dict from dep_urls_py.yaml
+  :return: boolean
+
+  It downloads files form urls to a temp directory first in order to avoid
+  to deal with any temp files. It helps keep clean final directory.
+  '''
+  if dep in configs.keys():
+    config = configs[dep]
+    dest_dir = '/'.join([license_dir, dep])
+    cur_temp_dir = 'temp_license_' + dep
+    os.mkdir(cur_temp_dir)
+    try:
+      is_file_available = False
+      # license is required, but not all dependencies have license.
+      # In case we have to skip, print out a message.
+      if config['license'] != 'skip':
+        wget.download(config['license'], cur_temp_dir + '/LICENSE')
+        is_file_available = True
+      # notice is optional.
+      if 'notice' in config:
+        wget.download(config['notice'], cur_temp_dir + '/NOTICE')
+        is_file_available = True
+      # copy from temp dir to final dir only when either file is abailable.
+      if is_file_available:
+        if os.path.isdir(dest_dir):
+          shutil.rmtree(dest_dir)
+        shutil.copytree(cur_temp_dir, dest_dir)
+      result = True
+    except Exception as e:
+      print('Error occurred when pull license from url.', 'dependency =',
+            dep, 'url =', config, 'error = ', e.decode('utf-8'))
+      result = False
+    finally:
+      shutil.rmtree(cur_temp_dir)
+      return result
+  else:
+    return False
+
+
+if __name__ == "__main__":
+  cur_dir = os.getcwd()
+  if cur_dir.split('/')[-1] != 'beam':
+    raise RuntimeError('This script should run from ~/beam directory.')
+  license_dir = os.getcwd() + '/licenses/python'
+  no_licenses = []
+
+  with open('licenses/scripts/dep_urls_py.yaml') as file:
+    dep_config = yaml.load(file)
+
+  install_pip_licenses()
+  dependencies = run_pip_licenses()
+  # add licenses for pip installed packages.
+  # try to pull licenses with pip-licenses tool first, if no license pulled,
+  # then pull from URLs.
+  for dep in dependencies:
+    if not (copy_license_files(dep) or pull_from_url(
+        dep['Name'].lower(), dep_config['pip_dependencies'])):
+      no_licenses.append(dep['Name'])
+
+  # add licenses for dependencies added to docker
+  for dep in dep_config['docker_dependencies']:
+    if not pull_from_url(dep, dep_config['docker_dependencies']):
+      no_licenses.append(dep)
+
+  # check if license are already pulled.
+  no_licenses = [
+      dep for dep in no_licenses
+      if not os.path.isdir(license_dir + '/' + dep.lower())
+  ]
+
+  if no_licenses:
+    py_version = '%d.%d' % (sys.version_info[0], sys.version_info[1])
+    raise RuntimeError(
```

Comment:
Can we add some instructions for the users what they should do when they see this error?